File: app12_kotlin_android/app/src/main/assets/code_samples/scala/supply-chain-dependency-integrity_defense.scala
// SECURE pattern
// WHY: an exact version means the build always resolves to a reviewed artifact, and sbt-dependency-check
// runs OWASP's SCA scanner against every build (mirrors this project's own composer audit + WPScan gate)
libraryDependencies += "org.example" %% "some-lib" % "2.3.1"

// project/plugins.sbt
addSbtPlugin("net.vonbuchholtz" % "sbt-dependency-check" % "5.1.0")

// build.sbt
dependencyCheckFailBuildOnCVSS := 7.0

File: app09_php_WORDPRESS/wp-content/plugins/securepress-2026/data/code_samples/java/supply-chain-dependency-integrity_defense.java
<!-- SECURE pattern -->
<!-- WHY: an exact, reviewed version pin means a build only ever resolves to a version the team has actually vetted -->
<dependency>
    <groupId>org.example</groupId>
    <artifactId>some-lib</artifactId>
    <version>1.4.2</version>
</dependency>
<!-- plus, in CI: mvn org.owasp:dependency-check-maven:check -->
